sched/jobDefSerializer_test.go

```diff
@@ -0,0 +1,249 @@
+}
+
+func Print(jobDef JobDefinition) {
+	for taskName, taskDef := range jobDef.Tasks {
+		fmt.Printf(fmt.Sprintf("taskName: %s\n", taskName))
+		fmt.Printf(fmt.Sprintf("\ttimeout: %s\n", taskDef.Timeout.String()))
+		fmt.Printf(fmt.Sprintf("\tsnapshotId: %s\n", taskDef.SnapshotId))
+		for envVarName, envVarVal := range taskDef.EnvVars {
+			fmt.Printf(fmt.Sprintf("\tenvVar:%s = %s\n", envVarName, envVarVal))
+		}
+		for i, arg := range taskDef.Argv {
+			fmt.Printf(fmt.Sprintf("\targ[%d]:%s\n", i, arg))
+		}
+	}
+	fmt.Printf("\n")
+}
+
+func Test_RandomSerializerDeserializer(t *testing.T) {
+	parameters := gopter.DefaultTestParameters()
+	parameters.MinSuccessfulTests = 1000
+	properties := gopter.NewProperties(parameters)
+
+	properties.Property("Serialize JobDef", prop.ForAll(
+		func(jobDef JobDefinition) bool {
+			//serialize then deserialized -  should get equal objects
+			if defAsByte, err := Serialize(&jobDef, serializeType); err != nil {
+				return false
+			} else {
+				deserializedJobDef, err := Deserialize(defAsByte, serializeType)
+				if err != nil {
+					return false
+				}
+				if ok, err1 := jobDef.Equal(*deserializedJobDef); !ok {
+					fmt.Printf(err1)
+					return false
+				}
+				//fmt.Printf(fmt.Sprintf("Job Def:%s\n",string(defAsByte)))
+				return true;
+			}
+
+		},
+
+		genJobDef(),
+	))
+
+	properties.TestingRun(t)
+
+}
+
+func genJobDef() gopter.Gen {
+	return func(genParams *gopter.GenParameters) *gopter.GenResult {
+		jobDef := genJobDefFromParams(genParams)
+		//Print(jobDef)
+		genResult := gopter.NewGenResult(jobDef, gopter.NoShrinker)
+		return genResult
+	}
+}
+
+func genJobDefFromParams(genParams *gopter.GenParameters) JobDefinition {
+	jobDef := JobDefinition{}
+	jobDef.JobType = fmt.Sprintf("jobType:%s",genString(genParams))
+
+	//number of tasks to run in this saga
+	numTasks := int(genParams.NextUint64() % 10)
+	var taskDefMap = make(map[string]TaskDefinition)
+
+	for i := 0; i < numTasks; i++ {
+		taskDef := TaskDefinition{}
+		taskName := fmt.Sprintf("taskName:%s", genString(genParams))
+		numArgs := int(genParams.NextUint64() % 5)
+
+		var j int
+		var args []string
+		for j = 0 ; j < numArgs; j++ {
+			args = append(args, fmt.Sprintf("arg%d:%s", j, genString(genParams)))
+			//taskDef.Argv = append(taskDef.Argv, fmt.Sprintf("arg%d:%s", j, genString(genParams)))
+		}
+		taskDef.Argv = args
+
+		envVarsMap :=  make(map[string]string)
+		numEnvVars := int(genParams.NextUint64() % 5)
+		for j = 0 ; j < numEnvVars; j++ {
+			envVarsMap[fmt.Sprintf("env%d",j)] = genString(genParams)
+		}
+		taskDef.EnvVars = envVarsMap
+
+		timeout, _ := time.ParseDuration(fmt.Sprintf("+%ds",genParams.NextUint64()))
+		taskDef.Timeout = timeout
+
+		taskDef.SnapshotId = fmt.Sprintf("snapShotId:%s",genString(genParams))
+		taskDefMap[taskName] = taskDef
+
+	}
+	jobDef.Tasks = taskDefMap
+
+	return jobDef
+}
+
+// generate a random string for the type
+func genString(genParams *gopter.GenParameters) string {
```

Comment:
We have this method in a common directory scoot/tests/testhelpers/generators.go  GenRandomAlphaNumericString() if you want to reuse that!